dhcpv4/ztpv4/ztp.go

```diff
@@ -15,14 +17,9 @@ type VendorData struct {
 
 var errVendorOptionMalformed = errors.New("malformed vendor option")
 
-// ParseVendorData will try to parse dhcp4 options looking for more
-// specific vendor data (like model, serial number, etc).
-func ParseVendorData(packet *dhcpv4.DHCPv4) (*VendorData, error) {
-	vc := packet.ClassIdentifier()
-	if len(vc) == 0 {
-		return nil, errors.New("vendor options not found")
-	}
+func parseClassIdentifier(packet *dhcpv4.DHCPv4) (*VendorData, error) {
 	vd := &VendorData{}
+	vc := packet.ClassIdentifier()
```

Comment:
NIT: can this be called in the switch?

I think it will be something like `switch vc := packet.ClassIdentifier(), vc {`